api/lib/mno_enterprise/concerns/controllers/jpi/v1/admin/organizations_controller.rb

```diff
@@ -22,22 +22,37 @@ module MnoEnterprise::Concerns::Controllers::Jpi::V1::Admin::OrganizationsContro
                               app_instances orga_invites users orga_relations
                               invoices credit_card demo_account main_address
                               current_credit].freeze
+
+    # Overwrite check support authorization for index action only.
+
+    # Workaround because using only and if are not possible (it checks to see if either satisfy, not both.)
+    # https://github.com/rails/rails/issues/9703#issuecomment-223574827
+    skip_before_action :block_support_users, if: -> { support_org_external_id && action_name == 'index'}
+    before_filter :support_enabled?, if: -> { support_org_external_id && action_name == 'index'}
+
+    skip_before_action :block_support_users, only: [:show]
+    before_filter :authorize_support_user_organization, only: [:show]
   end
 
   #==================================================================
   # Instance methods
   #==================================================================
   # GET /mnoe/jpi/v1/admin/organizations
   def index
-    if params[:terms]
+    if params[:organization_external_id]
```

Comment:
Could the support user manipulate the request to not pass the `params[:organization_external_id]`?
You might want to add a check on the support role?

Reply:
I think I took care of this:

1. If the request does not have `params[:organization_external_id]`, we will not skip the `block_support_users`, and the route will be blocked: : https://github.com/maestrano/mno-enterprise/pull/751/files#diff-3c6a54f1abe4b4fa1d7d79d37c5ae7f2R30. 

2. If we have `params[:organization_external_id]`, we check if current users are support and `Settings.admin_panel.support.enabled`